Add a couple of depiction helper functions and some JS bindings

ptosco opened the pull request:
- adds `normalizeDepiction()` and Python wrappers
- adds `straightenDepiction()` and Python wrappers
- adds an early check for R-labels to be in the accepted range to avoid throwing an exception later on
- adds a `kekulize` flag to `process_details()` to enable depicting molecules which fail to kekulize from JavaScript
- adds JavaScript functions to get fingerprints as `Uint8Array`
- adds JavaScript function to generate pickled molecule as `Uint8Array`
- adds JavaScript function to restore pickled molecule from `Uint8Array`
- adds `has_coords()` JavaScript function
- adds `set_new_coords()` to set JSMol coordinates in-place
- adds `get_mol_copy()` to obtain a JSMol copy

greglandrum commented on Code/GraphMol/Depictor/RDDepictor.h:
```diff
@@ -236,6 +236,59 @@ RDKIT_DEPICTOR_EXPORT void generateDepictionMatching3DStructure(
+  ARGUMENTS:
+  \param mol          - the molecule to be normalized
+  \param confId       - (optional) the id of the reference conformation to use
+  \param canonicalize - (optional) if 0 or 1, a canonical transformation is
+                        applied, If 0 (the default), the main molecule axis is
+                        aligned to the X axis.
+                        if 1, the canonical orientation is rotated by 90-degree
+                        to align the main molecule axis to the Y axis.
+                        If <0 or >1, no canonical transformation is applied.
```
I made the corresponding changes in my PR

greglandrum reviewed (changes requested):
Most of my suggestions are in the PR against your branch.

ptosco commented:
@greglandrum I should have addressed all of your remarks, plus the points we discussed on Slack. Please shout if I have inadvertently forgotten something.

greglandrum commented on Code/MinimalLib/tests/tests.js:
```diff
@@ -156,6 +171,21 @@ M  END`;
+    const kekRegExp = /  \d  \d  [12]  \d\n/g;
+    var mol = RDKitModule.get_mol("c1ccccc1");
+    var molblock = mol.get_molblock();
+    assert ([...molblock.matchAll(aromRegExp)].length === 0);
```
I really loathe javascript...

greglandrum reviewed (approved):
LGTM
Thanks Paolo!
